### DISCUSSION TITLE
Provide arguments for @requires fields

### DISCUSSION THREAD

**Discussion by KjellBerlin** [Netflix/dgs-framework#1868] [Q&A]
Hey all,

I have the following situation. 

ContactService
```
type Query {
    getContact (filters: Filters): Contact
}

type Contact {
    email: String
    address(filters: Filters): [Address] @external
    billingDetails: [BillingDetails] @requires(fields: "address { street postcode})}
}
```

AddressService
```
... everything address related
```

The problem is that here address behind @requires can not be resolved because it does not receive its filters. Is there any possibility to either

1.) Pass filters from query level to the address resolver, if this one is located in a different microservice? [Here](https://github.com/Netflix/dgs-framework/issues/36) the usage of `localContext` has been suggested. But as far as I understand it, the `localContext` is not being passed across service boundaries. So this would not work here
2.) Use exactly the same filters from the line above in the @requires field address like this?
```
type Contact {
    email: String
    address(filters: Filters): [Address] @external
    billingDetails(filters: Filters): [BillingDetails] @requires(fields: "address(filters: $filters) { street postcode})}
}
````

**Comment by srinivasankavitha** [comment]
The concept of local context is within your DGS. 

Are you using federation? Could you provide more details on what fields are resolved by which services? It sounds like getContact is from service A and address is resolved by serviceB? You will need also need an  @key field if you are using federation. The key is what will be passed as information to service B to look up the corresponding address - which sounds like should be the filters?

If not using federation, this set up is not going to work across multiple services.

**Reply by KjellBerlin** [reply]
Hi,
thank you for the fast reply!

Yes I am using federation, this is how my graph looks like at the moment in my working code (transformed to an address example). 

Service A
```
type Query {
    getContact (filters: Filters): Contact
}

type Contact @key(fields: "filters { contactId addressType }") {
    email: String
    filters: Filters
    address(filters: Filters): [Address] @external
    billingDetails: [BillingDetails] @requires(fields: "address { street postcode})}
}
```
Service B 
```
type Contact {
    filters: Filters @external
    address(filters: Filters): [Address]
}
```

To resolve `billingDetails`, `address` needs to be resolved. But address can only be resolved with filters. Here     
`billingDetails: [BillingDetails] @requires(fields: "address { street postcode})}`
I am not able to pass filters to the address resolver. Therefore I access the filters in Service B by adding the filters themselves as a property in ´Contact` at the moment, but this is not a very pretty solution in my opinion.

I am wondering now if there is different way to pass `filters` to the address resolver, something like this would be a dream for example:
`billingDetails: [BillingDetails] @requires(fields: "address(filters: $filters) { street postcode})}` 
But as far as I understand it, it is at the moment not possible to pass arguments like that.
